spinoso-time/src/time/tzrs/offset.rs

```diff
@@ -0,0 +1,273 @@
+        match self {
+            Self::Utc => TimeZoneRef::utc(),
+            Self::Fixed(local_time_types) => match TimeZoneRef::new(&[], local_time_types, &[], &None) {
+                Ok(tz) => tz,
+                Err(_) => GMT,
+            },
+
+            Self::Tz(zone) => *zone,
+        }
+    }
+}
+
+impl From<&str> for Offset {
+    /// Construct a Offset based on the [accepted MRI values]
+    ///
+    /// Accepts:
+    ///
+    /// - `[+/-]HH[:]MM`
+    /// - A-I representing +01:00 to +09:00
+    /// - K-M representing +10:00 to +12:00
+    /// - N-Y representing -01:00 to -12:00
+    /// - Z representing 0 offset
+    ///
+    /// [accepted MRI values]: https://ruby-doc.org/core-2.6.3/Time.html#method-c-new
+    #[inline]
+    #[must_use]
+    fn from(input: &str) -> Self {
+        match input {
+            "A" => Self::fixed(1),
+            "B" => Self::fixed(2),
+            "C" => Self::fixed(3),
+            "D" => Self::fixed(4),
+            "E" => Self::fixed(5),
+            "F" => Self::fixed(6),
+            "G" => Self::fixed(7),
+            "H" => Self::fixed(8),
+            "I" => Self::fixed(9),
+            "K" => Self::fixed(10),
+            "L" => Self::fixed(11),
+            "M" => Self::fixed(12),
+            "N" => Self::fixed(-1),
+            "O" => Self::fixed(-2),
+            "P" => Self::fixed(-3),
+            "Q" => Self::fixed(-4),
+            "R" => Self::fixed(-5),
+            "S" => Self::fixed(-6),
+            "T" => Self::fixed(-7),
+            "U" => Self::fixed(-8),
+            "V" => Self::fixed(-9),
+            "W" => Self::fixed(-10),
+            "X" => Self::fixed(-11),
+            "Y" => Self::fixed(-12),
+            "Z" => Self::utc(),
+            _ => {
+                lazy_static! {
+                    static ref HH_MM_MATCHER: Regex = Regex::new(r"^([\-\+]{1})(\d{2})(\d{2})$").unwrap();
+                }
+                if HH_MM_MATCHER.is_match(input) {
+                    let caps = HH_MM_MATCHER.captures(input).unwrap();
+
+                    let sign = if caps.get(1).unwrap().as_str() == "+" { 1 } else { -1 };
+                    let hours = caps.get(2).unwrap().as_str().parse::<i32>().unwrap();
+                    let minutes = caps.get(3).unwrap().as_str().parse::<i32>().unwrap();
+
+                    let offset_seconds: i32 = sign * ((hours * SECONDS_IN_HOUR) + (minutes * SECONDS_IN_MINUTE));
+                    Self::fixed(offset_seconds)
+                } else {
+                    // TODO: ArgumentError
+                    Self::utc()
+                }
+            }
+        }
+    }
+}
+
+impl From<String> for Offset {
+    fn from(input: String) -> Self {
+        Offset::from(input.as_str())
+    }
+}
+
+impl From<TimeZoneRef<'static>> for Offset {
+    #[inline]
+    #[must_use]
+    fn from(tz: TimeZoneRef<'static>) -> Self {
+        Self::tz(tz)
+    }
+}
+
+impl From<i32> for Offset {
```

Comment:
Are all `i32` valid offsets or should this be a `TryFrom` impl?

Do the invariants of the `Self::Fixed` variant assume the `i32` is bounded?

Reply:
Hopefully answered in another comment above (e.g. range is +/- 24 * 60 * 60), but will address in errors PR since that has the Errors we want to throw.